From 19f21ce1d86ebe9f409ad03dfd15fa92bbae5f89 Mon Sep 17 00:00:00 2001
From: Yongxin Liu <yongxin.liu@windriver.com>
Date: Tue, 24 Sep 2024 20:14:56 +0000
Subject: [PATCH] qat20: limit the use of adf_enable/disable_aer()

adf_enable/disable_aer() was dropped in kernel commit 197cccc771e
("crypto: qat - drop redundant adf_enable_aer()"), so we also drop
it in out-of-tree driver of qat for kernel >= 6.3.

Upstream-Status: Inappropriate [Code released in tarball form only]

Signed-off-by: Yongxin Liu <yongxin.liu@windriver.com>
---
 .../qat/drivers/crypto/qat/qat_common/adf_aer.c        | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/quickassist/qat/drivers/crypto/qat/qat_common/adf_aer.c b/quickassist/qat/drivers/crypto/qat/qat_common/adf_aer.c
index ba277e0..9255b94 100644
--- a/quickassist/qat/drivers/crypto/qat/qat_common/adf_aer.c
+++ b/quickassist/qat/drivers/crypto/qat/qat_common/adf_aer.c
@@ -378,10 +378,15 @@ int adf_enable_aer(struct adf_accel_dev *accel_dev, struct pci_driver *adf)
 #else
 	adf->err_handler = &adf_err_handler;
 #endif
+
+#if (KERNEL_VERSION(6, 3, 0) <= LINUX_VERSION_CODE)
+	return 0;
+#else
 	ret = pci_enable_pcie_error_reporting(pdev);
 	if (ret)
 		dev_warn(&pdev->dev,
 			 "QAT: Failed to enable AER, error code %d\n", ret);
+#endif
 
 	return 0;
 }
@@ -400,8 +405,11 @@ EXPORT_SYMBOL_GPL(adf_enable_aer);
 void adf_disable_aer(struct adf_accel_dev *accel_dev)
 {
 	struct pci_dev *pdev = accel_to_pci_dev(accel_dev);
-
+#if (KERNEL_VERSION(6, 3, 0) <= LINUX_VERSION_CODE)
+	return;
+#else
 	pci_disable_pcie_error_reporting(pdev);
+#endif
 }
 EXPORT_SYMBOL_GPL(adf_disable_aer);
 
-- 
2.44.1

